unittests/AST/IndexSubsetTests.cpp

```diff
@@ -210,3 +210,119 @@ TEST(IndexSubset, FindPrevious) {
   EXPECT_EQ(indices1->findPrevious(/*endIndex*/ 1), 0);
   EXPECT_EQ(indices1->findPrevious(/*endIndex*/ 0), -1);
 }
+
+TEST(IndexSubset, Lowering) {
+  TestContext testCtx;
+  auto &C = testCtx.Ctx;
+  // ((T, T)) -> ()
+  EXPECT_EQ(
+      autodiff::getLoweredParameterIndices(
+          IndexSubset::get(C, 1, {0}),
+          FunctionType::get({
+              FunctionType::Param(
+                  TupleType::get({C.TheAnyType, C.TheAnyType}, C))},
+              C.TheEmptyTupleType)),
+      IndexSubset::get(C, 2, {0, 1}));
+  // ((), (T, T)) -> ()
+  EXPECT_EQ(
+      autodiff::getLoweredParameterIndices(
+          IndexSubset::get(C, 2, {1}),
+          FunctionType::get({
+              FunctionType::Param(C.TheEmptyTupleType),
+              FunctionType::Param(
+                  TupleType::get({C.TheAnyType, C.TheAnyType}, C))},
+                                 C.TheEmptyTupleType)),
+      IndexSubset::get(C, 2, {0, 1}));
+  // (T, (T, T)) -> ()
+  EXPECT_EQ(
+    autodiff::getLoweredParameterIndices(
+      IndexSubset::get(C, 2, {1}),
+      FunctionType::get({
+          FunctionType::Param(C.TheAnyType),
+          FunctionType::Param(
+            TupleType::get({C.TheAnyType, C.TheAnyType}, C))},
+        C.TheEmptyTupleType)),
+    IndexSubset::get(C, 3, {1, 2}));
+  // (T, (T, T)) -> ()
+  EXPECT_EQ(
+    autodiff::getLoweredParameterIndices(
+      IndexSubset::get(C, 2, {0, 1}),
+      FunctionType::get({
+          FunctionType::Param(C.TheAnyType),
+          FunctionType::Param(
+            TupleType::get({C.TheAnyType, C.TheAnyType}, C))},
+        C.TheEmptyTupleType)),
+    IndexSubset::get(C, 3, {0, 1, 2}));
+  // (T, ((T, T)), (T, T), T) -> ()
+  EXPECT_EQ(
+    autodiff::getLoweredParameterIndices(
+      IndexSubset::get(C, 4, {0, 1, 3}),
+      FunctionType::get({
+          FunctionType::Param(C.TheAnyType),
+          FunctionType::Param(
+              TupleType::get({
+                  TupleType::get({C.TheAnyType, C.TheAnyType}, C)}, C)),
+          FunctionType::Param(
+            TupleType::get({C.TheAnyType, C.TheAnyType}, C)),
+          FunctionType::Param(C.TheAnyType)},
+        C.TheEmptyTupleType)),
+    IndexSubset::get(C, 6, {0, 1, 2, 5}));
+  // Method (T) -> ((T, T), (T, T), T) -> ()
+  // TODO(TF-874): Fix this unit test.
```

Comment:
Note: I have WIP for fixing TF-874 and uncommenting this unit test. It's a bit invasive, so it probably makes sense to land after AutoDiff upstreaming is done.